pandas/core/groupby/generic.py

```diff
@@ -251,6 +251,8 @@ def aggregate(self, func=None, *args, **kwargs):
 
             try:
                 return self._python_agg_general(func, *args, **kwargs)
+            except AssertionError:
```

Comment:
Hmm what's the point of the bare raise here? I get needed to differentiate now from the `except Exception`, but coupled with change above is there something actually throwing the `AssertionError`? Or do we mean to be catching `ValueError` here?

Reply:
> but coupled with change above is there something actually throwing the AssertionError

No, which means that during development we can now add assertions and not have them get swallowed.